tests/integration/datadict/submission/test_endpoints.py

```diff
@@ -1030,3 +1030,236 @@ def test_zero_decimal_float(client, pg_driver, cgci_blgsp, submitter):
+        ]
+        == None
+    )
+    assert (
+        json.loads(resp.data)["entities"][0]["properties"]["indels_identified"] == None
+    )
+
+
+def test_update_to_null_invalid(client, pg_driver, cgci_blgsp, submitter):
+    """
+    Test that updating a required field to null results in an error
+    """
+    headers = submitter
+    data = json.dumps(
+        {
+            "type": "experiment",
+            "submitter_id": "BLGSP-71-06-00019",
+            "projects": {"id": "daa208a7-f57a-562c-a04a-7a7c77542c98"},
+        }
+    )
+    resp = client.put(BLGSP_PATH, headers=headers, data=data)
+    assert resp.status_code == 200, resp.data
+    id = json.loads(resp.data)["entities"][0]["id"]
+
+    data = json.dumps({"submitter_id": None,})
+    resp = client.put(BLGSP_PATH, headers=headers, data=data)
+    assert resp.status_code == 400, resp.data
+
+    data = json.dumps({"type": None,})
+    resp = client.put(BLGSP_PATH, headers=headers, data=data)
+    assert resp.status_code == 400, resp.data
+
+    data = json.dumps({"id": None,})
+    resp = client.put(BLGSP_PATH, headers=headers, data=data)
+    assert resp.status_code == 400, resp.data
+
+    resp = client.get(
+        f"/v0/submission/CGCI/BLGSP/entities/{id}", headers=headers, data=data,
+    )
+    print(json.dumps(json.loads(resp.data), indent=4, sort_keys=True))
+    assert (
+        json.loads(resp.data)["entities"][0]["properties"]["submitter_id"]
+        == "BLGSP-71-06-00019"
+    )
+    assert json.loads(resp.data)["entities"][0]["properties"]["type"] == "experiment"
+    assert json.loads(resp.data)["entities"][0]["properties"]["id"] == id
+
+
+def test_update_to_null_valid_tsv(client, pg_driver, cgci_blgsp, submitter):
+    """
+    Test that we can update a TSV file with null
+    """
+
+    data = json.dumps(
+        {
+            "type": "experiment",
+            "submitter_id": "BLGSP-71-06-00019",
+            "projects": {"id": "daa208a7-f57a-562c-a04a-7a7c77542c98"},
+            "experimental_description": "my desc",
+            "number_samples_per_experimental_group": 1,
+        }
+    )
+
+    headers = submitter
+    resp = client.put(BLGSP_PATH, headers=headers, data=data)
+    print(json.dumps(json.loads(resp.data), indent=4, sort_keys=True))
+    assert resp.status_code == 200, resp.data
+    resp = client.get(
+        f"/v0/submission/CGCI/BLGSP/entities/{json.loads(resp.data)['entities'][0]['id']}",
+        headers=headers,
+        data=data,
+    )
+    print(json.dumps(json.loads(resp.data), indent=4, sort_keys=True))
+
+    data = {
+        "type": "experiment",
+        "submitter_id": "BLGSP-71-06-00019",
+        "projects.id": "daa208a7-f57a-562c-a04a-7a7c77542c98",
+        "experimental_description": None,
+        "number_samples_per_experimental_group": None,
+    }
+
+    # convert to TSV (save to file)
+    file_path = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), "data/experiment_tmp.tsv"
+    )
```

Comment:
so in the resulting file, the `None` are empty values right?

Reply:
Yes